src/codeu/chat/client/simplegui/ChatGuiFX.java

```diff
@@ -392,9 +441,34 @@ private void getAllMessages(ConversationSummary conversation) {
             final String authorName = clientContext.user.getName(m.author);
 
             final String displayString = String.format("%s: [%s]: %s",
-                ((authorName == null) ? m.author : authorName), m.creation, m.content);
+                ((authorName.isEmpty()) ? m.author : authorName), m.creation, m.content);
 
             messageList.addAll(displayString);
       }
     }
+
+    /**
+    * Populates the list of users to be displayed
+    * @param users  the current list of users that will be cleared & updated
+    */
+    private void fillUserList(ListView<String> users) {
+        clientContext.user.updateUsers();
+        users.getItems().clear();
+
+        for (final User u : clientContext.user.getUsers()) {
+            usersList.addAll(u.name);
+        }
+    }
+
+    // TODO: set up a loop to where this updates every half sec or so
+    /**
+    * When the user presses the update button, update the GUI by clearing & and filling
```

Comment:
Just curious: why do you need to clear the GUI and refill it? What if I have a million conversations going, or a conversation with a million messages? Is there a way to update without clearing everything every time?